pkg/conversion/capi2mapi/powervs_fuzz_test.go

```diff
@@ -0,0 +1,170 @@
+			conversiontest.CAPIMachineFuzzerFuncs(powerVSProviderIDFuzzer, powerVSMachineKind, capibmv1.GroupVersion.String(), infra.Status.InfrastructureName),
+			powerVSMachineFuzzerFuncs,
+		)
+	})
+
+	Context("PowerVSMachineSet Conversion", func() {
+
+		fromMachineSetAndPowerVSMachineTemplateAndPowerVSCluster := func(machineSet *capiv1.MachineSet, infraMachineTemplate client.Object, infraCluster client.Object) capi2mapi.MachineSetAndMachineTemplate {
+			powerVSMachineTemplate, ok := infraMachineTemplate.(*capibmv1.IBMPowerVSMachineTemplate)
+			Expect(ok).To(BeTrue(), "input infra machine template should be of type %T, got %T", &capibmv1.IBMPowerVSMachineTemplate{}, infraMachineTemplate)
+
+			powerVSCluster, ok := infraCluster.(*capibmv1.IBMPowerVSCluster)
+			Expect(ok).To(BeTrue(), "input infra cluster should be of type %T, got %T", &capibmv1.IBMPowerVSCluster{}, infraCluster)
+
+			return capi2mapi.FromMachineSetAndPowerVSMachineTemplateAndPowerVSCluster(machineSet, powerVSMachineTemplate, powerVSCluster)
+		}
+
+		conversiontest.CAPI2MAPIMachineSetRoundTripFuzzTest(
+			scheme,
+			infra,
+			infraCluster,
+			&capibmv1.IBMPowerVSMachineTemplate{},
+			mapi2capi.FromPowerVSMachineSetAndInfra,
+			fromMachineSetAndPowerVSMachineTemplateAndPowerVSCluster,
+			conversiontest.ObjectMetaFuzzerFuncs(capiNamespace),
+			conversiontest.CAPIMachineFuzzerFuncs(powerVSProviderIDFuzzer, powerVSMachineKind, capibmv1.GroupVersion.String(), infra.Status.InfrastructureName),
+			conversiontest.CAPIMachineSetFuzzerFuncs(powerVSTemplateKind, capibmv1.GroupVersion.String(), infra.Status.InfrastructureName),
+			powerVSMachineFuzzerFuncs,
+			powerVSMachineTemplateFuzzerFuncs,
+		)
+	})
+})
+
+func powerVSProviderIDFuzzer(c fuzz.Continue) string {
+	//Power VS provider id format: ibmpowervs://<region>/<zone>/<service_instance_id>/<instance_id>
+	return fmt.Sprintf("ibmpowervs://tok/tok04/%s/%s", strings.ReplaceAll(c.RandString(), "/", ""), strings.ReplaceAll(c.RandString(), "/", ""))
+}
+
+func powerVSMachineFuzzerFuncs(codecs runtimeserializer.CodecFactory) []interface{} {
+	return []interface{}{
+		func(serviceInstance *capibmv1.IBMPowerVSResourceReference, c fuzz.Continue) {
+			switch c.Int31n(3) {
+			case 0:
+				serviceInstance.ID = ptr.To(c.RandString())
+			case 1:
+				serviceInstance.Name = ptr.To(c.RandString())
+			case 3:
+				serviceInstance.RegEx = ptr.To(c.RandString())
+			}
+		},
+		func(network *capibmv1.IBMPowerVSResourceReference, c fuzz.Continue) {
+			switch c.Int31n(3) {
+			case 0:
+				network.ID = ptr.To(c.RandString())
+			case 1:
+				network.Name = ptr.To(c.RandString())
+			case 3:
+				network.RegEx = ptr.To(c.RandString())
+			}
+		},
+		func(imageRef *corev1.LocalObjectReference, c fuzz.Continue) {
+			imageRef.Name = c.RandString()
+
+		},
+		func(spec *capibmv1.IBMPowerVSMachineSpec, c fuzz.Continue) {
+			c.FuzzNoCustom(spec)
+
+			spec.ServiceInstanceID = ""
+			spec.Image = nil
```

Comment:
Why clearing these?

Reply:
ServiceInstanceID is deprecated, Image id identified by another filed ImageRef

Reply:
Might be worth noting those here so we don't forget, assuming these will never be changed

Reply:
Added a comment